kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/security/ElytronUserGroupAdapter.java

```diff
@@ -36,14 +43,56 @@
         if (identityOptional.isPresent()) {
             SecurityIdentity identity = identityOptional.get();
             if (identity.getPrincipal().getName().equals(userId)) {
-                return StreamSupport.stream(identity.getRoles().spliterator(), false)
-                        .collect(Collectors.toList());
+                return toRoles(identity);
+            } else {
+                SecurityDomain securityDomain = SecurityDomain.getCurrent();
+                if (securityDomain == null) {
+                    return Collections.emptyList();
+                }
+
+                String runAsPrincipal = userId;
+                SecurityIdentity newIdentity;
+                try {
+                    if (! runAsPrincipalExists(securityDomain, runAsPrincipal)) {
+                        newIdentity = securityDomain.createAdHocIdentity(runAsPrincipal);
```

Comment:
Is this really needed? Because `createAdHocIdentity` creates `an empty ad-hoc identity. The identity will have no authorization information` so roles will be empty, wdyt?

Reply:
done

Reply:
removed